File: pat1058.py
A,B = input().split()

g1,s1,k1 = (int(x) for x in A.split('.'))
g2,s2,k2 = (int(x) for x in B.split('.'))
result = [g1 + g2,s1 + s2,k1 + k2]
if result[2] >= 29:
    result[1] += 1
    result[2] -= 29

if result[1] >= 17:
    result[0] += 1
    result[1] -= 17

result = [str(x) for x in result]   #   将数组每个元素转化为字符形式
print('.'.join(result))

File: pat1058.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String[] line = reader.readLine().split(" ");
        System.out.println(addInHogwarts(line[0], line[1]));
    }

    static String addInHogwarts(String A, String B) {
        String[] numA = A.split("\\.");
        String[] numB = B.split("\\.");
        int[] result = new int[3];
        // 将各位相加
        int knut = Integer.parseInt(numA[2]) + Integer.parseInt(numB[2]);
        int sickle = Integer.parseInt(numA[1]) + Integer.parseInt(numB[1]);
        int galleon = Integer.parseInt(numA[0]) + Integer.parseInt(numB[0]);
        if (knut >= 29) {
            // knut位有进位情况
            knut -= 29;
            sickle += 1;
        }
        if (sickle >= 17) {
            // sickle位有进位情况
             sickle -= 17;
            galleon += 1;
        }

        return galleon + "." + sickle + "." + knut;
    }

}
